server/handlers/image_handler.py

```diff
@@ -1,19 +1,53 @@
+import logging
+import socket
 from random import random, randint, choice
 import string
 
 
 class ImageHandler:
-    def __init__(self, gs):
+    def __init__(self, gs, config):
+        self.logger = logging.getLogger("main")
         self.gs = gs
+        self.config = config
+        print("╠ CREATED IMAGE HANDLER")
+        self.logger.info("╠ CREATED IMAGE HANDLER")
 
+    def socket_connect(self):
+        while True:  # Restart the socket connection on disconnect
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                s.bind((self.config["uav"]["images"]["host"], self.config["uav"]["images"]["port"]))
+                s.listen()
+                conn, addr = s.accept()
+                with conn:
+                    self.logger.critical("[Image] Socket connection established with %s:%s", *addr)
+                    curr_image = ""
+                    while True:
+                        resp = conn.recv(4096)
+                        if not resp:
+                            break
+                        resp = resp.decode("utf-8")
+                        loc = resp.find(" ")
+                        if loc != -1:
+                            curr_image += resp[:loc]
+                            res = self.gs.image.process_image(curr_image)
+                            curr_image = resp[loc + 1:]
+                            self.logger.debug("[Image] Successfully retreived image from socket")
+                            if res:
+                                self.logger.info("[Image] Successfully identified ODLC from Image")
+                            conn.sendall(bytes("Image Complete", "utf-8"))
+                        else:
+                            curr_image += resp
+                    self.logger.critical("[Image] Lost socket connection with %s:%s", *addr)
```

Comment:
I think this should be logged as `warning`, or `error`.

Reply:
Same reasoning as the other `CRITICAL` log.

Reply:
It's different from the other logging statement though. The philosophy behind the `warning` level, at least in the article I linked, was that warnings were events that could potentially become an error. I believe this fits that description. In fact, you could have it emit an error event when it loses connection long enough for the definition of lost comms in Auvsi Suas.